ecr-login/api/client.go

```diff
@@ -16,47 +16,70 @@ package api
 import (
 	"encoding/base64"
 	"fmt"
+	"net/url"
 	"regexp"
 	"strings"
 	"time"
 
-	"github.com/pkg/errors"
-
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ecr"
-	"github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cache"
+	"github.com/aws/aws-sdk-go/service/ecrpublic"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+
+	"github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cache"
+)
+
+const (
+	proxyEndpointScheme = "https://"
+	programName         = "docker-credential-ecr-login"
+	ecrPublicName       = "public.ecr.aws"
+	ecrPublicEndpoint   = proxyEndpointScheme + ecrPublicName
 )
 
-const proxyEndpointScheme = "https://"
-const programName = "docker-credential-ecr-login"
+var ecrPattern = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(\-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?$`)
```

Comment:
`\-fips` -> `-fips`? `-` is not a meta character.

Reply:
Done